File: oogiri-frontend/src/types/game.ts
export interface Game {
  id: string;
  status: 'waiting' | 'in_progress' | 'finished';
  playerIds: string[];
  currentRound: number;
  totalRounds: number;
  theme: string;
  answers: Answer[];
}

export interface Player {
  id: string;
  name: string;
  score: number;
}

export interface Answer {
  id: string;
  content: string;
  playerId: string;
}

export interface JoinGameRequest {
  gameId: string;
  username: string;
}

File: oogiri-backend/src/models/game.go
package models

type Game struct {
    ID          string   `firestore:"id"`
    Name        string   `firestore:"name"`
    Description string   `firestore:"description"`
    CreatedAt   int64    `firestore:"created_at"`
    UpdatedAt   int64    `firestore:"updated_at"`
    Status      string   `firestore:"status"`       // 'waiting', 'in_progress', 'finished'
    PlayerIDs   []string `firestore:"playerIds"`
    CurrentRound int     `firestore:"current_round"`
    TotalRounds  int     `firestore:"total_rounds"`
    Theme       string   `firestore:"theme"`
    Answers     []Answer `firestore:"answers"`
}

type Player struct {
    ID       string `firestore:"id"`
    Username string `firestore:"username"` // 追加
    Score    int    `firestore:"score"`
}

type Answer struct {
    ID       string `firestore:"id"`
    Content  string `firestore:"content"`
    PlayerID string `firestore:"player_id"`
}